homeassistant/components/onkyo/config_flow.py

```diff
@@ -165,6 +167,49 @@ async def async_step_eiscp_discovery(
             ),
         )
 
+    async def async_step_ssdp(
+        self, discovery_info: ssdp.SsdpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle flow initialized by SSDP discovery."""
+        _LOGGER.debug("Config flow start ssdp: %s", discovery_info)
+
+        if udn := discovery_info.ssdp_udn:
+            udn_parts = udn.split(":")
+            if len(udn_parts) == 2:
+                uuid = udn_parts[1]
+                last_uuid_section = uuid.split("-")[-1].upper()
+                await self.async_set_unique_id(last_uuid_section)
+                self._abort_if_unique_id_configured()
+
+        if discovery_info.ssdp_location is None:
+            _LOGGER.error("SSDP location is None")
+            return self.async_abort(reason="unknown")
+
+        host = URL(discovery_info.ssdp_location).host
+
+        if host is None:
+            _LOGGER.error("SSDP host is None")
+            return self.async_abort(reason="unknown")
+
+        try:
+            info = await async_interview(host)
+        except OSError:
+            _LOGGER.exception("Unexpected exception interviewing host %s", host)
+            return self.async_abort(reason="unknown")
+
+        if info is None:
+            _LOGGER.debug("SSDP eiscp is None: %s", host)
+            return self.async_abort(reason="cannot_connect")
+
+        await self.async_set_unique_id(info.identifier)
+        self._abort_if_unique_id_configured()
```

Comment:
```suggestion
        self._abort_if_unique_id_configured(updates={CONF_HOST: self._receiver_info.host})
```

Likely `async_step_eiscp_discovery` we can update the host address if it changes so the user doesn't have to reconfig manually

Reply:
ruff may complain about the formatting of this suggestion

Reply:
What if the user did manually enter it previously? How do we differentiate it? Because overriding that in the background would probably be surprising and I don't think is a good idea.

Reply:
I mean, when we are at this point we've established that the device with the same identifier as back then is now at a new host, and we even connected to it. So I think it is safe to say that the manually entered address has been changed

Reply:
Not necessarily. User may have manually entered a hostname, like "my-receiver.internal" or "onkyo-tr-nr7000.local", etc. This is actually preferable to a raw IP address that we get from discovery here.
And even if it was an IP address, it was still directly entered by the user. Maybe the discovery we see here is only temporary or something. I would still consider that an UX antipattern, to change things like that in the background. But at least here it can be argued I guess, with hostname it is obviously not something the user would want to happen under any circumstances.

I'm now just realizing though, that we do exactly that in many other integrations already. I think this should be discussed and rethinked, as it can potentially break some users setup.

Reply:
We prefer IP addresses for anything that can be discovered and auto updated via a stable identifier that does not require TLS or name verification as it eliminates: DNS, local name collisions, changes to naming scheme between router firmware or hardware replacement, or other name change issue as points of failure. It's a long established architecture decision.  If we want to change that it should get an architecture discussion.

If the integration does some type of name or TLS verification the config entry update is usually implemented manually to exclude updating from name to IP addresses (example in unifiprotect)

Reply:
> What it introduces instead is the reliance on those finicky multicast discovery protocols, which in my experience are the very first ones to break in case of any network changes (or even without one).

This project only supports host networking and we generally don't go out of way to add complexity to handle broken network setups as it makes integrations a lot more complex and harder to maintain.

> I mean, you can have a user who manually enters a FQDN and has a properly set up DNS, and now you're overwriting it (without even informing the user!) with some random IP.

If this integration wants to implement more complex logic to handle FQDNs differently, and there is a good reason to do so similar to what `unifiprotect` does that would be fine to do in another PR without an architecture discussion.  If there is nothing special here, that it should get an arch discussion before implementing it.

Reply:
>This project only supports host networking and we generally don't go out of way to add complexity to handle broken network setups as it makes integrations a lot more complex and harder to maintain.

Yes, but in this case it can be Home Assistant itself which inadvertently messes up the setup.

>If this integration wants to implement more complex logic to handle FQDNs differently, and there is a good reason to do so similar to what unifiprotect does that would be fine to do in another PR without an architecture discussion. If there is nothing special here, than it should get an arch discussion before implementing it.

No, nothing special here, we should do the same as everywhere else. If it is to be changed, it should be applied everywhere and enforced into the future.

I only initially responded, because I wasn't aware that this is a common practice.